src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.TrailingHeaders.cs

```diff
@@ -0,0 +1,225 @@
+                            "\r\n"));
+
+                    using (HttpResponseMessage response = await getResponseTask)
+                    {
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        Assert.Contains("chunked", response.Headers.GetValues("Transfer-Encoding"));
+                        Assert.Contains("MyCoolTrailerHeader", response.Headers.GetValues("Trailer"));
+
+                        // Pending read on the response content.
+                        Assert.Empty(response.TrailingHeaders);
+
+                        Stream stream = await response.Content.ReadAsStreamAsync();
+                        Byte[] data = new Byte[100];
+                        int start = 0;
+                        await stream.ReadAsync(data, start, 4);
+                        start = 4;
+
+                        // Intermedia test - haven't reached stream EOF yet.
+                        Assert.Empty(response.TrailingHeaders);
+                        Assert.Contains("data", System.Text.Encoding.Default.GetString(data));
+
+                        bool streamEOFReached = false;
+                        while (!streamEOFReached)
+                        {
+                            if (stream.Read(data, start, 10) == 0)
+                            {
+                                streamEOFReached = true;
+                            }
+                            start += 10;
+                        }
+
+                        Assert.Contains("amazingtrailer", response.TrailingHeaders.GetValues("MyCoolTrailerHeader"));
+                        Assert.Contains("World", response.TrailingHeaders.GetValues("Hello"));
+                    }
+                }
+            });
+        }
+
+        [Fact]
+        public async Task GetAsync_ForbiddenTrailingHeaders_Throws()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(async url =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    await Assert.ThrowsAsync<HttpRequestException>(() => client.GetAsync(url));
+                }
+            }, server => server.AcceptConnectionSendCustomResponseAndCloseAsync(
+                "HTTP/1.1 200 OK\r\n" +
+                "Connection: close\r\n" +
+                "Transfer-Encoding: chunked\r\n" +
+                "Trailer: Set-Cookie, MyCoolTrailerHeader, Hello, Content-Type\r\n" +
+                "\r\n" +
+                "4\r\n" +
+                "data\r\n" +
+                "0\r\n" +
+                "Set-Cookie: yummy\r\n" +
+                "MyCoolTrailerHeader: amazingtrailer\r\n" +
+                "Hello: World\r\n" +
+                "Content-Type: text\r\n" +
+                "\r\n"));
+        }
+
+        [Fact]
+        public async Task GetAsync_MissingTrailer_TrailingHeadersAccepted()
```

Comment:
Is this test needed at all?  It seems like the `GetAsyncDefaultCompletionOption_TrailingHeaders_Available(false)` test covers this already.

Reply:
In this case the header was not advertised via Trailer: header. So this covers the "unexpected trailing header" scenario.

Reply:
> In this case the header was not advertised via Trailer: header. So this covers the "unexpected trailing header" scenario.

In GetAsyncDefaultCompletionOption_TrailingHeaders_Available, with the false argument, it's not advertised via the Trailer header either: that's what the `false` argument does:
https://github.com/dotnet/corefx/pull/35337/files#diff-2db0187c9cf2e8514bbf995c553c999fR35

Reply:
yes, you are right @stephentoub . I removed GetAsync_MissingTrailer_TrailingHeadersAccepted.